src/resource-heap.cpp

```
@@ -0,0 +1,52 @@
+    if (requirements.size == 0 || requirements.alignment == 0)
+        return SLANG_E_INVALID_ARG;
+    if ((placement.offset % requirements.alignment) != 0)
+        return SLANG_E_INVALID_ARG;
+
+    const ResourceHeapDesc& heapDesc = placement.heap->getDesc();
+    if (heapDesc.memoryType != requirements.memoryType)
+        return SLANG_E_INVALID_ARG;
+    if (!isResourceHeapKindCompatible(heapDesc.kind, requirements.heapKind))
+        return SLANG_E_INVALID_ARG;
+    if (placement.offset > heapDesc.size || requirements.size > heapDesc.size - placement.offset)
+        return SLANG_E_INVALID_ARG;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)